x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -749,17 +758,38 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return MVT_SOURCE_LAYER_NAME;
   }
 
+  _errorUpdatingIndex(numIndexes: number) {
```

Comment:
Thanks for adding toast. There a few problems with the fix
1) message needs to be wrapped in i18n wrapper since its user facing
2) message should distinguish between no matching patterns and too manh matching patterns
3) Caller of addFeature and deleteFeature should be creating the toast exception. Imagine another source implemented addFeature and deleteFeature. we would want to handle the toasts in a single place and not the source. Also caller needs to handle exception so there is not an uncaught error

Reply:
:+1: to 1 and 2. Partial agreement with 3:

> Imagine another source implemented addFeature and deleteFeature. we would want to handle the toasts in a single place and not the source.

I believe we're handling this in the correct place. If we leverage these same functions elsewhere in `es_search_source`, I would expect the same limitations and behavior. If we broaden these functions to be leveraged by other sources, then we should consider how similar behavior is between sources, if we want to move this up a level, etc.

Reply:
The toast handling should be handled by the caller, which in this case is the action handlers. That way the UI is the same across all sources. @thomasneirynck thoughts?

Reply:
discussed offline. Makes sense to move this out of the `es_search_source`, which feels too low-level. Higher-up seems a better fit to raise toasts.

Agreed that this is somewhat of a judgement call, but in general, the `*_source.ts` code is focussed on the mechanics of fetching data, and less concerned with the UX.

Reply:
I've updated the code to catch the error and trigger the toast in `map_actions`. If we want to treat source and related files as strictly business logic, we should also consider moving [this toast logic](http://github.com/aaronjcaldwell/kibana/blob/7dbc9b8bf8063042e560379a061f6e4e7c11c72c/x-pack/plugins/maps/public/classes/sources/es_search_source/util/load_index_settings.ts#L56-L56) as well, however that's beyond the scope of this bug fix.